File: lib/models/appointment.dart

import 'dart:convert';

class Appointment {
  String? id;
  DateTime dateTime;
  double listedPrice;
  double percentDiscount;
  double fullPrice;
  bool? filled;
  String? serviceOfferedId;
  String? officeId;
  String? patientFilledId;

  Appointment ({
    required this.id,
    required this.dateTime,
    required this.listedPrice,
    required this.percentDiscount,
    required this.fullPrice,
    required this.filled,
    this.serviceOfferedId,
    this.officeId,
    this.patientFilledId
  });

  void appointmentFromDatabase(var id, var dateTime, var listedPrice, var percentDiscount, var fullPrice, var filled, var serviceOfferedId, var officeId, var patientFilledId){
      this.id = id;
      this.dateTime = DateTime.parse(dateTime);
      this.listedPrice = double.parse(listedPrice);
      this.fullPrice = double.parse(fullPrice);
      this.percentDiscount = double.parse(percentDiscount);
      if(filled.toLowerCase()=="true"){
        this.filled = true;
      }else{
        this.filled = false;
      }
      this.serviceOfferedId = serviceOfferedId;
      this.officeId = officeId;
      this.patientFilledId = patientFilledId;
  }
}

File: python/db_objects/appointment.py
class Appointment:
    def __init__(self, id, time, listed_price, full_price, filled, practice_id, patient_id, service_id, is_active):
        self.id = id
        self.time = time
        self.listed_price = listed_price
        self.full_price = full_price
        self.filled = filled
        self.practice_id = practice_id
        self.patient_id = patient_id
        self.service_id = service_id
        self.is_active = is_active